packages/material-ui/src/Grow/Grow.test.js

```diff
@@ -83,6 +91,14 @@ describe('<Grow />', () => {
         wrapper.setProps({
```

Comment:
It's not apparent what this should accomplish. Either set it when mounting or add some "before" assertions so that it's obvious what this is supposed to do.

Same for the other `setProps` steps. If this creates redundant assertions (due to other tests) we can consolidate but don't have to. The description of the test cases holds already close to zero information so we have to be a bit more verbose with the code.

Reply:
Looking at the test, we make sure the transition hooks are correctly called. We used to have test logic that wasn't testing the whole event propagation (`wrapper.find(Transition).props().onExited()`).

Reply:
Yes but this test isn't actually testing what the prop change does. It's possible that all of these hooks are already called when mounting.

Reply:
that setProps just puts it into an exiting state, I'm unsure what I should test before that state? I can consolidate the entering and exiting tests into one test?

Reply:
I like what you did with the in/out tests: https://github.com/mui-org/material-ui/blob/7499902fdc0269b1c8c8d3ad49bd437dd0c159c6/packages/material-ui/src/Collapse/Collapse.test.js#L89.